--- ksymoops-2.4.9/ksyms.c.orig	2002-03-01 01:50:26.000000000 +0100
+++ ksymoops-2.4.9/ksyms.c	2003-08-10 00:37:57.000000000 +0200
@@ -146,8 +146,9 @@
     RE_COMPILE(&re_ksyms,
 	"^([0-9a-fA-F]{4,})"		/* 1 address */
 	" +"				/* white space */
-	"([^ ]+)"			/* 2 symbol */
-	"( +\\[([^ ]+)\\])?$",		/* 3,4 space, module (optional) */
+	"([^ ] +)?"			/* 2 symbol section(?) (Linux 2.6) */
+	"([^ ]+)"			/* 3 symbol */
+	"( +\\[([^ ]+)\\])?$",		/* 4,5 space, module (optional) */
 	REG_NEWLINE|REG_EXTENDED,
 	&re_ksyms_pmatch);
 
@@ -158,9 +159,9 @@
 	return;
 
     re_strings(&re_ksyms, line, re_ksyms_pmatch, &string);
-    if (strncmp(string[2], "GPLONLY_", 8) == 0)
-	strcpy(string[2], string[2]+8);
-    add_ksyms(string[1], ' ', string[2], string[4], options);
+    if (strncmp(string[3], "GPLONLY_", 8) == 0)
+	strcpy(string[3], string[3]+8);
+    add_ksyms(string[1], ' ', string[3], string[5], options);
     re_strings_free(&re_ksyms, &string);
 }
 
